### DISCUSSION TITLE
Ways to contribute

### DISCUSSION THREAD

**Discussion by jrgfogh** [erez-strauss/lockfree_mpmc_queue#3] [Ideas]
Great [presentation at CppCon!](https://youtu.be/M3v2GfeGJYs)

In one of the slides, you mention some "next steps," which are:

1. Benchmarks - compare with similar queues / disruptor
2. Porting to other platforms Windows, RISC-V, Arm
3. Blocking using conditional variable - non-busy wait on empty or full
4. Unify API, according to the WG21 concurrent queue paper
5. Conflation implementation
6. Performance improvement
7. Multiple read of the same entry - reads count

Are these still relevant? How would one go about contributing?
I would be interested in looking into porting to Windows and/or ARM.

**Comment by erez-strauss** [comment]
Hi Jørgen,

Thank you for your interest.
Yes, the porting work is very much needed - thank you.

1. Benchmarks - there are few in the source tree, but more are needed, including comparing with other queues, including result visualization.

2. The porting to windows needs to overcome the issue of how to represent a 16 bytes entities and call the 16 bytes CAS atomically.
It will also require some work on the CMake.

3., 4., 6., 7. contributions are welcome.

5. is available using the 
`bool exchange(index_type& i, value_type old_value, value_type new_value)`
method, please see the test example code.

In order to contribute, please fork the repository, make your changes to your fork-repository and place a Pull-Request.
( https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project )

Thank you, Erez
